samples/cpp/live_detect_qrcode.cpp

```diff
@@ -0,0 +1,177 @@
+#include "opencv2/objdetect.hpp"
+#include "opencv2/imgproc.hpp"
+#include "opencv2/highgui.hpp"
+#include <string>
+#include <iostream>
+
+using namespace std;
+using namespace cv;
+
+void getMatWithQRCodeContour(Mat &color_image, vector<Point> transform, int fps = -1);
+int  liveQRCodeDetect();
+int  showImageQRCodeDetect(string mode, string in, string out);
+
+int main(int argc, char *argv[])
+{
+    const string keys =
+        "{h help ? |        | print help messages }"
+        "{m mode   | camera | detect mode (supported: camera, show_image, save_image) }"
```

Comment:
the logic of the sample is weird. Please, change it to the following:
* no parameters - use `camera` mode.
* `i` is set - `show_image`
* both `i` and `o` are set - `save_image`.